mantidproject/mantid
# Add option to specify reflection type to PredictFractionalPeaks

**Description of work.**



**To test:**

Download the data from #26910. You'll need to NeXus from the .zip file. The following script gives an idea of how to use the algorithm.

```python
Load(Filename='unindexed-peaks.nxs', OutputWorkspace='sx_peaks')
LoadIsawUB(InputWorkspace='sx_peaks', Filename='pr-26807.txt')
PredictFractionalPeaks(Peaks='sx_peaks', Hmin=-1, Hmax=1, Kmin=-1, Kmax=1, Lmin=-1, Lmax=1, ReflectionCondition='All-face centred', RequirePeaksOnDetector=False, FracPeaks='fractional')
LoadEmptyInstrument(InstrumentName='WISH', OutputWorkspace='wish')
```

Once the `fractional` workspace has been created open the instrument view on the `wish` workspace and drag the `fractional` workspace on to it. In this example, because `RequirePeaksOnDetector=False` then you should see peaks not on a detector bank. Flip this option to `True` and the peaks should all be on a detector.

Play with the ranges and if you expand them more peaks should be found and vice versa. Selecting the another reflection type should change the peaks found. 

Fixes #26690

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
